pkg/certificate/auto/provider.go

```diff
@@ -45,16 +45,16 @@ func New(c client.Client) interfaces.Provider {
 
 var userAutoConfig *interfaces.Config
 
-func (ac *Provider) EnsureCertificateSecret(ctx context.Context, secretName, namespace string,
+func (ac *Provider) EnsureCertificateSecret(ctx context.Context, secretKey client.ObjectKey,
 	cfg *interfaces.Config) error {
 	// Save the user defined AutoConfig so that it can be returned from GetCertificateConfig
 	userAutoConfig = cfg
 
 	var secret corev1.Secret
-	err := ac.Get(ctx, client.ObjectKey{Name: secretName, Namespace: namespace}, &secret)
+	err := ac.Get(ctx, secretKey, &secret)
 	if err != nil {
 		if k8serrors.IsNotFound(err) {
-			err := ac.createNewSecret(ctx, secretName, namespace, cfg)
+			err := ac.createNewSecret(ctx, secretKey.Name, secretKey.Namespace, cfg)
```

Comment:
I think we can update `createNewSecret` to use `client.ObjectKey` as well

Reply:
Yes, let me fix that. Thank you.